### PR TITLE
Support ICMPv4 match/set action and fix a typo

### PR DESCRIPTION
**Pull request by commandgjj**
1, support ICMPv4 match/set action;
2, fix a typo "SLL" -> "TLL" in NdTll match action.

Co-authored-by: Linan Zheng
Signed-off-by: Linan Zheng
Signed-off-by: Gao Jinjun <gjinjun@gmail.com>

### CODE REVIEW

**Comment by commandgjj** [comment]
@wenyingd @hangyan @tnqn Can you help review and merge this PR? Thanks!

**Comment by hangyan** [comment]
LGTM

File: ofctrl/ofAction.go [hunk 1]
```diff
@@ -686,3 +688,31 @@ func (a *SetICMPv6CodeAction) GetActionMessage() openflow13.Action {
 func (a *SetICMPv6CodeAction) GetActionType() string {
 	return ActTypeSetICMP6Code
 }
+
+type SetICMPv4TypeAction struct {
+	Type uint8
+}
+
+func (a *SetICMPv4TypeAction) GetActionMessage() openflow13.Action {
+	field, _ := openflow13.FindFieldHeaderByName("NXM_OF_ICMP_TYPE", false)
```
Comment (tnqn) [inline ofctrl/ofAction.go:697]: I notice `SetICMPv6TypeAction`'s `GetActionMessage` uses lower case for the field name, is it another typo or it's not case sensitive?

Comment (commandgjj) [reply ofctrl/ofAction.go:697]: Yes. The field name is not case sensitive. The FindFieldHeaderByName() will convert field name to upper automaticly.

**Review by tnqn — APPROVED** [review]
LGTM

**Review by wenyingd — COMMENTED** [review]
LGTM. One kind remind, ofnet currently supports to two ways to generate the FlowMod message, 1) functiion GenerateFlowModMessage, called by Flow.Next(FgraphElem) 2) function generateFlowMessage, called by Flow.Send(operationType). If you only use 2nd function, the code change in the PR is enough, but if you still need to call 1st function, you should also add other function calls like SetICMPType/SetICMPCode, and maintain the action type is Flow.flowActions, then change function "installFlowActions"

**Comment by commandgjj** [comment]
> 

Thanks for kind reminding, Wenying. Currently, we only used Flow.Send() function to generate flow message. I think this PR should be enough for our case. 

And I checked the Flow.Next() and installFlowActions(), we may need to add ICMP Type/Code related NewICMPXxxxField to operate ICMP field in libOpenflow too. Let's do this in separated patch if we need to use Flow.Next() in future. Thanks!

**Comment by wenyingd** [comment]
> > 
> 
> Thanks for kind reminding, Wenying. Currently, we only used Flow.Send() function to generate flow message. I think this PR should be enough for our case.
> 
> And I checked the Flow.Next() and installFlowActions(), we may need to add ICMP Type/Code related NewICMPXxxxField to operate ICMP field in libOpenflow too. Let's do this in separated patch if we need to use Flow.Next() in future. Thanks!

OK, then could you add comments or TODO before SetICMPv4XXXAction about it, so that we could avoid other callers making mistakes?

**Review by wenyingd — APPROVED** [review]
LGTM

**Comment by commandgjj** [comment]
> LGTM

Wenying, can you please help merge this PR if no any other problem? Thanks!

**Pull request merged**